pkg/handlers/aws/mutation/instancetype/inject_control_plane.go

```diff
@@ -0,0 +1,133 @@
+	"github.com/d2iq-labs/capi-runtime-extensions/common/pkg/capi/clustertopology/variables"
+	capav1 "github.com/d2iq-labs/capi-runtime-extensions/common/pkg/external/sigs.k8s.io/cluster-api-provider-aws/v2/api/v1beta2"
+	awsclusterconfig "github.com/d2iq-labs/capi-runtime-extensions/pkg/handlers/aws/clusterconfig"
+	"github.com/d2iq-labs/capi-runtime-extensions/pkg/handlers/generic/clusterconfig"
+)
+
+const (
+	// VariableName is the external patch variable name.
+	VariableName = "instanceType"
+
+	// WorkerHandlerNamePatch is the name of the inject handler.
+	ControlPlaneHandlerNamePatch = "AWSInstanceTypeControlPlanePatch"
+)
+
+type awsInstanceTypeControlPlanePatchHandler struct {
+	variableName      string
+	variableFieldPath []string
+}
+
+var (
+	_ commonhandlers.Named     = &awsInstanceTypeControlPlanePatchHandler{}
+	_ mutation.GeneratePatches = &awsInstanceTypeControlPlanePatchHandler{}
+	_ mutation.MetaMutator     = &awsInstanceTypeControlPlanePatchHandler{}
+)
+
+func NewControlPlaneMetaPatch() *awsInstanceTypeControlPlanePatchHandler {
+	return newAWSInstanceTypeControlPlanePatchHandler(
+		clusterconfig.MetaVariableName,
+		clusterconfig.MetaControlPlaneConfigName,
+		awsclusterconfig.AWSVariableName,
+		VariableName,
+	)
+}
+
+func newAWSInstanceTypeControlPlanePatchHandler(
+	variableName string,
+	variableFieldPath ...string,
+) *awsInstanceTypeControlPlanePatchHandler {
+	return &awsInstanceTypeControlPlanePatchHandler{
+		variableName:      variableName,
+		variableFieldPath: variableFieldPath,
+	}
+}
+
+func (h *awsInstanceTypeControlPlanePatchHandler) Name() string {
+	return ControlPlaneHandlerNamePatch
+}
+
+func (h *awsInstanceTypeControlPlanePatchHandler) Mutate(
+	ctx context.Context,
+	obj runtime.Object,
+	vars map[string]apiextensionsv1.JSON,
+	holderRef runtimehooksv1.HolderReference,
+	_ client.ObjectKey,
+) error {
+	log := ctrl.LoggerFrom(ctx).WithValues(
+		"holderRef", holderRef,
+	)
+
+	instanceTypeVar, found, err := variables.Get[v1alpha1.InstanceType](
+		vars,
+		h.variableName,
+		h.variableFieldPath...,
+	)
+	if err != nil {
+		return err
+	}
+	if !found {
+		log.V(5).Info("AWS instance type variable for control-plane not defined")
+		return nil
+	}
+
+	log = log.WithValues(
+		"variableName",
+		h.variableName,
+		"variableFieldPath",
+		h.variableFieldPath,
+		"variableValue",
+		instanceTypeVar,
+	)
+
+	return patches.Generate(
+		obj,
+		vars,
+		&holderRef,
+		selectors.InfrastructureControlPlaneMachines(
+			"v1beta2",
+			"AWSMachineTemplate",
+		),
+		log,
+		func(obj *capav1.AWSMachineTemplate) error {
+			log.WithValues(
+				"patchedObjectKind", obj.GetObjectKind().GroupVersionKind().String(),
+				"patchedObjectName", client.ObjectKeyFromObject(obj),
+			).Info("setting instance type in control plane AWSMachineTemplate spec")
+
+			obj.Spec.Template.Spec.InstanceType = string(instanceTypeVar)
+
+			return nil
+		},
+	)
+}
+
+func (h *awsInstanceTypeControlPlanePatchHandler) GeneratePatches(
```

Comment:
this method is only useful helper in unit tests since the meta handler will directly call the `Mutate` func of its registered handlers. It does not hurt to keep it here though.

Reply:
It also needed to satisfy this interface https://github.com/d2iq-labs/capi-runtime-extensions/blob/8fad4c739699d216f4ef6e62d0ebf014ccb39516/common/pkg/capi/clustertopology/handlers/mutation/interfaces.go#L19-L25

Reply:
I think `GeneratePatches` interface does not need to satisfy at this level. it just has to be satisfied at the meta handler level.